### DISCUSSION TITLE
Method SmartDate.SetDefaultFormatString() not working as expected.

### DISCUSSION THREAD

**Discussion by adrianwright109** [MarimerLLC/csla#1973] [Questions] [answered]
I have a .NET Core 3.1 app which is using CSLA version `5.3.1`.  I have a couple of properties using the `SmartDate` type.  I would like to be able to globally set the default date format to `"yyyy-MM-dd"`, so that I don't have to set it every time I create a new instance of the `SmartDate` type, e.g. `new SmartDate(DateTime.Today) { FormatString = "yyyy-MM-dd" }`.

I have tried setting it in `Startup.cs` class after the call to `services.AddCsla();` in the `ConfigureServices` method.

The code I have used is `SmartDate.SetDefaultFormatString("yyyy-MM-dd");`

```
obj.StartDate = new SmartDate(new DateTime(2020, 12, 31));
```
When using the `SmartDate` in a string interpolation expression for example `$"SmartDate: {obj.StartDate}"`.
Expected result: `"SmartDate: 2020-12-31"`
Actual result: `"SmartDate: 31/12/2020"`

Is it possible to set the default date format for `SmartDate` globally in .NET Core ?

**Comment by rockfordlhotka** [comment, marked as answer]
Yes. The default default is `d`, but you can set the default globally during app startup. For example:

```c#
  Csla.SmartDate.SetDefaultFormatString("d");
```

Alternately, you can change the Windows date display. The reason `d` is the default is because it is the OS-defined short date format, and should automatically change based on the user's culture settings.

**Reply by rockfordlhotka** [reply]
Oh, doh! I see that you're already setting the default format 😳

**Reply by rockfordlhotka** [reply]
This might require a little research, as string interpolation didn't exist when this was all first written 🙂

But it seems to me like it should work, so the fact that it doesn't is puzzling.

It could be [here in the string conversion](https://github.com/MarimerLLC/csla/blob/7677b83d068f4e708df6b722c6bb0abba95608c8/Source/Csla/SmartDate.cs#L1333).

Or in the [ToString implementation](https://github.com/MarimerLLC/csla/blob/7677b83d068f4e708df6b722c6bb0abba95608c8/Source/Csla/SmartDate.cs#L456).

Both of those call the [Text property implementation](https://github.com/MarimerLLC/csla/blob/7677b83d068f4e708df6b722c6bb0abba95608c8/Source/Csla/SmartDate.cs#L399) though. And that implementation does use the format string.

```c#
    public string Text
    {
      get { return DateToString(this.Date, FormatString, _emptyValue); }
      set { this.Date = StringToDate(value, _emptyValue); }
    }
```

Yeah, some research is probably necessary to figure out which path is used by string interpolation, and why the `Text` property isn't generating the result.

**Reply by rockfordlhotka** [reply]
Guess I shouldn't have moved this to be a discussion... 😢

**Reply by adrianwright109** [reply]
Sorry, my mistake it is working correctly with string interpolation.  Please close issue.
